config/initializers/yabeda.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+return if Settings.scc.try(:metrics) && (Settings.scc.metrics.try(:enabled) && (!!Settings.scc.metrics.enabled  == false))
+
+# require 'yabeda'
+# require 'yabeda/rails'
+# require 'yabeda/sidekiq'
+# require 'yabeda/prometheus'
+
+# Configure prometheus client
+Prometheus::Client.config.data_store = Prometheus::Client::DataStores::DirectFileStore.new(
+  dir: './tmp/prometheus/'
+)
+
+# Configure yabeda
+Yabeda.configure do
+  assign_labels = lambda {
+    default_tag :environment, Rails.env
+    default_tag :application, 'rmt'
+  }
+
+  group :sidekiq, &assign_labels
+  group :rails, &assign_labels
+  group :rake, &assign_labels
```

Comment:
For a PoC I'd first leave it until here (line 23). First collect all metrics that the framework gives you, and then incorporate more metrics later.

The reasoning for this is that the metric reporting is still a bit rudimentary, as an implementor you have to shift the context from the place of invocation to the event listener (for example, the one in line 57), and THEN account for the metric. While this is fine, it's not reload-safe, if you're going to rely on reloading this flow won't work.

Among other things this is sadly by design in rails, event subscriber registries are not reloadable.